src/client/Microsoft.Identity.Client.Brokers/WamAdapters.cs

```diff
@@ -0,0 +1,236 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client.Core;
+using Microsoft.Identity.Client.Internal.Requests;
+using Microsoft.Identity.Client.OAuth2;
+using Microsoft.Identity.Client.Utils;
+using static Microsoft.Identity.Client.Broker.RuntimeBroker;
+
+namespace Microsoft.Identity.Client.Broker
+{
+    internal class WamAdapters
+    {
+        private const string WamErrorPrefix = "WAM Error ";
+        
+        //MSA-PT Auth Params
+        private const string NativeInteropMsalRequestType = "msal_request_type";
+        private const string ConsumersPassthroughRequest = "consumer_passthrough";
+
+        //MSAL Runtime Error Response 
+        private enum ResponseStatus : int
+        {
+            Unexpected = 0,
+            Reserved = 1,
+            InteractionRequired = 2,
+            NoNetwork = 3,
+            NetworkTemporarilyUnavailable = 4,
+            ServerTemporarilyUnavailable = 5,
+            ApiContractViolation = 6,
+            UserCanceled = 7,
+            ApplicationCanceled = 8,
+            IncorrectConfiguration = 9,
+            InsufficientBuffer = 10,
+            AuthorityUntrusted = 11,
+            UserSwitch = 12,
+            AccountUnusable = 13,
+            UserDataRemovalRequired = 14
+        };
+
+        /// <summary>
+        /// Create WAM Error Response
+        /// </summary>
+        /// <param name="authResult"></param>
+        /// <param name="authenticationRequestParameters"></param>
+        /// <param name="logger"></param>
+        /// <returns></returns>
+        /// <exception cref="MsalClientException"></exception>
+        /// <exception cref="MsalUiRequiredException"></exception>
+        /// <exception cref="MsalServiceException"></exception>
+        internal static MsalException CreateWamErrorResponse(
+            NativeInterop.AuthResult authResult,
+            AuthenticationRequestParameters authenticationRequestParameters,
+            ICoreLogger logger)
+        {
+            MsalServiceException serviceException = null;
+            string internalErrorCode = null;
+            string errorMessage;
+            int errorCode;
+
+            switch ((int)authResult.Error.Status)
+            {
+                case (int)ResponseStatus.UserCanceled:
+                    logger.Verbose($"[WamBroker] {MsalError.AuthenticationCanceledError} {MsalErrorMessage.AuthenticationCanceled}");
```

Comment:
For all the error scenarios I would suggest to do logger.Error

Reply:
This is consistent with what we are doing currently. Please re-activate if you disagree.

Reply:
Yes, this is consistent with current code but I am working on https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/3278 to resolve the issue where we have some failure and unable to repro or figure it out since the apps have MSAL logs enabled for Error/ Warning level and providing verbose logs is not always preferred by them.

Reply:
updated to error

Reply:
Disregard above since it's been updated.